components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/APIManagerTenantSyncListener.java

```diff
@@ -0,0 +1,371 @@
+    }
+
+    @Override
+    public void onTenantInitialActivation(int tenantId) throws StratosException {
+        // Not implemented.
+    }
+
+    @Override
+    public void onTenantActivation(int tenantId) throws StratosException {
+
+        if (canCreateTenantInAPIM(tenantId)) {
+            TenantInfoBean tenantInfo = new TenantInfoBean();
+            try {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+                RealmService realmService = IdentityMgtServiceComponent.getRealmService();
+                String domain = realmService.getTenantManager().getTenant(tenantId).getDomain();
+
+                tenantInfo.setTenantDomain(domain);
+                tenantInfo.setTenantId(tenantId);
+                tenantInfo.setActive(true);
+                sendEvent(tenantInfo, TenantManagement.ACTION_ACTIVATE, TenantManagement.EVENT_ACTIVATE_TENANT_URI);
+            } catch (UserStoreException e) {
+                log.error("Error while activating tenant in API Manager ", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        }
+    }
+
+    @Override
+    public void onTenantDeactivation(int tenantId) throws StratosException {
+
+        if (canCreateTenantInAPIM(tenantId)) {
+            TenantInfoBean tenantInfo = new TenantInfoBean();
+            try {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+                RealmService realmService = IdentityMgtServiceComponent.getRealmService();
+                String domain = realmService.getTenantManager().getTenant(tenantId).getDomain();
+
+                tenantInfo.setTenantDomain(domain);
+                tenantInfo.setTenantId(tenantId);
+                tenantInfo.setActive(false);
+                sendEvent(tenantInfo, TenantManagement.ACTION_DEACTIVATE, TenantManagement.EVENT_ACTIVATE_TENANT_URI);
+            } catch (UserStoreException e) {
+                log.error("Error while deactivating tenant in API Manager ", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        }
+    }
+
+    @Override
+    public void onSubscriptionPlanChange(int tenentId, String oldPlan, String newPlan) throws StratosException {
+        // Not implemented.
+    }
+
+    @Override
+    public int getListenerOrder() {
+
+        return 0;
+    }
+
+    @Override
+    public void onPreDelete(int tenantId) throws StratosException {
+        // Not implemented.
+    }
+
+    /**
+     * Check whether tenant sharing is enabled and whether the creation is only a root organization.
+     * 
+     * @param tenantId id of the tenant
+     * @return boolean
+     */
+    private boolean canCreateTenantInAPIM(int tenantId) {
```

Comment:
we should check only whether tenant creation firing is enabled or not

Reply:
Change these logs https://github.com/wso2/carbon-identity-framework/pull/7075/files#diff-9d4e1778ea285830b2bf85655d94b9eb936720063456eddd2e014f555fed5818R214 and existing class names with APIM reference

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/7075/commits/a6be893041c5d4a898b3d2f74d8525409740cf5c